cmd/stackdriver-prometheus-sidecar/main_test.go

```diff
@@ -38,31 +35,12 @@ import (
 var promPath string
 
 func TestMain(m *testing.M) {
-	flag.Parse()
-	if testing.Short() {
+	if os.Getenv("RUN_MAIN") == "" {
+		// Normal test mode.
 		os.Exit(m.Run())
 	}
-	// On linux with a global proxy the tests will fail as the go client(http,grpc) tries to connect through the proxy.
-	os.Setenv("no_proxy", "localhost,127.0.0.1,0.0.0.0,:")
-
-	var err error
-	promPath, err = os.Getwd()
-	if err != nil {
-		fmt.Printf("can't get current dir :%s \n", err)
-		os.Exit(1)
-	}
-	promPath = filepath.Join(promPath, "prometheus")
-
-	build := exec.Command("go", "build", "-o", promPath)
-	output, err := build.CombinedOutput()
-	if err != nil {
-		fmt.Printf("compilation error :%s \n", output)
-		os.Exit(1)
-	}
 
-	exitCode := m.Run()
-	os.Remove(promPath)
-	os.Exit(exitCode)
+	main()
```

Comment:
Based on https://golang.org/pkg/testing/#hdr-Main, I think we should call `os.Exit(main())`.

Reply:
`main` doesn't return a value. My understanding is that `os.Exit` must only be called in the direct test path. Where are you reading otherwise?